File: asgn2/python_code/task4_java_submit/DumbFixture.java
package assg2task4;
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author kenchan
 */
public class DumbFixture extends Fixture {
    protected Attendant person_in_charge;
    protected Robot robot_in_charge;
    protected int is_robot=0;
    public DumbFixture(String Name, double Weight, int Position, Attendant Person_in_charge){
        super(Name, Weight, Position);
        this.person_in_charge = Person_in_charge;
        this.is_robot=0;
    }
    public DumbFixture(String Name, double Weight, int Position, Robot Robot_in_charge){
        super(Name, Weight, Position);
        this.robot_in_charge = Robot_in_charge;
        this.is_robot=1;
    }
    public void MoveTo(Room Destination){
        if(is_robot==0){this.person_in_charge.CarryTo(this, Destination);}
        else{this.robot_in_charge.CarryTo(this, Destination);}
    }
    public void AssignTo(Attendant Person_in_charge){
        this.person_in_charge = Person_in_charge;
    }
    public void AssignTo(Robot Robot_in_charge){
        this.robot_in_charge = Robot_in_charge;
    }
}

File: Programming_Languages/asgn2/python_code/task3_python/DumbFixture.py
#this is the python dump Fixture


import Fixture
import Attendant
import Room

class DumbFixture(Fixture):
    def __init__(self, Name, Weight, Position, Person_in_charge):
        Fixture.__init__(self, Name, Weight, Position)
        self._person_in_charge = Person_in_charge

    def MoveTo(self, Destination):
        self._person_in_charge.CarryTo(self, Destination)

    def AssignTo(self,Person_in_charge):
        self._person_in_charge=Person_in_charge
